packages/amplify-provider-awscloudformation/lib/amplify-service-migrate.js

```diff
@@ -0,0 +1,110 @@
+  if (!amplifyClient) {
+    // This happens when the Amplify service is not available in the region
+    const message = `Amplify service is not available in the region ${awsConfig.region ? awsConfig.region : ''}`;
+    context.print.error(message);
+    throw new Error(message);
+  }
+
+  const hasPermission = await checkAmplifyServiceIAMPermission(context, amplifyClient);
+  if (!hasPermission) {
+    const message = 'Permssions to access Amplify service is required.';
+    context.print.error(message);
+    throw new Error(message);
+  }
+
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify && inputParams.amplify.appId) {
+    const amplifyAppId = inputParams.amplify.appId;
+
+    try {
+      const getAppResult = await amplifyClient
+        .getApp({
+          appId: amplifyAppId,
+        })
+        .promise();
+      context.print.info(`Amplify AppID found: ${amplifyAppId}. Amplify App name is: ${getAppResult.app.name}}`);
+    } catch (e) {
+      context.print.error(
+        `Amplify AppID: ${amplifyAppId} not found. Please ensure your local profile matches the AWS account or region in which the Amplify app exists.`
+      );
+      context.print.info(e);
+      throw e;
+    }
+
+    let backendEnvs = [];
+    let listEnvResponse = {};
+    do {
+      listEnvResponse = await amplifyClient
+        .listBackendEnvironments({
+          appId: amplifyAppId,
+          nextToken: listEnvResponse.nextToken,
+        })
+        .promise();
+
+      backendEnvs = backendEnvs.concat(listEnvResponse.backendEnvironments);
+    } while (listEnvResponse.nextToken);
+
+    const { envName } = projectDetails.localEnvInfo;
+    const { StackName, DeploymentBucketName } = projectDetails.teamProviderInfo[envName][constants.ProviderName];
+    if (!backendEnvs.includes(envName)) {
+      context.print.info(`Adding backend environment ${envName} to AWS Amplify Console app: ${amplifyAppId}`);
+      const createEnvParams = {
+        appId: amplifyAppId,
+        environmentName: envName,
+        stackName: StackName,
+        deploymentArtifacts: DeploymentBucketName,
+      };
+      await amplifyClient.createBackendEnvironment(createEnvParams).promise();
+    } else {
+      // If env is already in the app, verify the the stack name match
+      const getEnvParams = {
+        appId: amplifyAppId,
+        environmentName: envName,
+      };
+      const { backendEnvironment } = await amplifyClient.getBackendEnvironment(getEnvParams).promise();
+      if (StackName !== backendEnvironment.stackName) {
+        const message = `Stack name mistmatch for the backend environment ${envName}. Local: ${StackName}, Amplify: ${backendEnvironment.stackName}`;
+        context.print.error(message);
+        throw new Error(message);
+      }
+    }
+
+    // Add the appId to the meta data and team provider info and write the files
+    const { amplifyMeta, teamProviderInfo } = projectDetails; // context.exeInfo;
```

Comment:
What's the "// context.exeInfo;" comment?

Reply:
removed